src/session-pool.ts

```diff
@@ -772,7 +778,12 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
       );
     }
 
-    if (!this.isFull) {
+    // Only create a new session if there are more waiters than sessions already
+    // being created.
+    if (
+      !this.isFull &&
+      this._pending + this._pendingPrepare <= this._numWaiters
```

Comment:
Shouldn't we change `<=` to `<`? When `this._pending + this._pendingPrepare == this._numWaiters`, should we create a new session?

Reply:
The current requester is at this point not included in `this._numWaiters`, so if we already have `this._pending === this._numWaiters`, the current requester will not be served by any of the pending sessions. So the check here should be `<=`.